### PR TITLE
Bug fixes1

### PR DESCRIPTION
**Pull request by au1994**
@vaidik
@vishesh92
@dash1291

### CODE REVIEW

File: outermysite/polls/imdbScraper.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    release_date = title_details.find(
+        "meta",
+        {"itemprop": "datePublished"}
+    )
+
+    genre = title_details.find_all("span", {"itemprop": "genre"})
+
+    json_object = {}
+
+    if (title is not None):
```
Comment (dash1291) [inline outermysite/polls/imdbScraper.py]: don't need parens here and elsewhere

File: outermysite/polls/views.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
+from django.shortcuts import render
```
Comment (dash1291) [inline outermysite/polls/views.py]: you should read on how to arrange imports properly

File: outermysite/polls/imdbScraper.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    except requests.exceptions.RequestException as e:
+        logging.info(e)
+        raise SystemExit(e)
+
+    soup = BeautifulSoup(r.content, "html.parser")
+    movie_details = soup.find("div", {"class": "plot_summary"})
+    json_array = []
+
+    if movie_details is None:
+        print "No results found"
```
Comment (dash1291) [inline outermysite/polls/imdbScraper.py:59]: you should return HTTP status code 404 in this case in the API response

File: outermysite/polls/imdbScraper.py [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+        json_object = {}
+        json_object['Title'] = title
+        json_object['Url'] = url
+        json_object['Id'] = Id
+        json_array.append(json_object)
+
+    return json_array
+
+
+def getMovieResults(Id):
```
Comment (vishesh92) [inline outermysite/polls/imdbScraper.py]: Variable naming. `Id` should be `id`.

Comment (dash1291) [inline outermysite/polls/imdbScraper.py]: also, function names should be snake case - `get_movie_results`

File: outermysite/polls/imdbScraper.py [hunk 5]
```diff
@@ -0,0 +1,118 @@
+from bs4 import BeautifulSoup
```
Comment (dash1291) [inline outermysite/polls/imdbScraper.py]: please rearrange

File: outermysite/polls/imdbScraper.py [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+        return json_object
+
+    for item in movies:
+        if item is not None:
+            title = item.get_text().strip()
+            url = item.find("a").get("href")
+            movie_id = url[7:16]
+        url = "http://www.imdb.com" + url
+        json_object = {}
+        json_object['Title'] = title
```
Comment (vishesh92) [inline outermysite/polls/imdbScraper.py]: JSON keys should be in camel case.
https://google.github.io/styleguide/jsoncstyleguide.xml#Property_Name_Guidelines

File: README.md [hunk 7]
```diff
@@ -1 +1,5 @@
 # Django_imdbApi
+
+Search api - http://192.168.5.79:8080/imdb/search?q=fight&page=3
```
Comment (dash1291) [inline README.md:3]: you can change this to your heroku url

File: outermysite/polls/views.py (hunk 2 above)
Comment (dash1291) [inline outermysite/polls/views.py]: this grouping of imports is not perfect. `requests` could go together with bs4 and django imports. `imdbScraper` can go in the end

File: outermysite/polls/imdbScraper.py (hunk 3 above)
Comment (dash1291) [inline outermysite/polls/imdbScraper.py:59]: any specific reason why `print` is left here?

File: outermysite/polls/urls.py [hunk 8]
```diff
@@ -0,0 +1,8 @@
+from django.conf.urls import url
+
+from . import views
+
+urlpatterns = [
+        url(r'^search/', views.search_movie, name='search_movie'),
+        url(r'^exact/', views.exact_movie, name='exact_movie'),
```
Comment (dash1291) [inline outermysite/polls/urls.py:7]: you could make your URLs look better doing some tweaks to make them look more like REST APIs,

See an example of something similar to what you want to do - https://dev.twitter.com/rest/reference/get/users/search

**Pull request merged**